avocado/core/job.py

```diff
@@ -474,6 +477,9 @@ def _run(self):
             stacktrace.log_exc_info(sys.exc_info(), 'avocado.app.debug')
             self._remove_job_results()
             raise exceptions.OptionValidationError(details)
+
+        self.job_pre_post_dispatcher.map_methods('pre', self)
```

Comment:
OK so the pre script is executed before we are checking for valid test_suite. It adds additional requirement for plugins as they might need to deal with that, but I think it's the right place as one might want to eg. receive the email about non-executed test suite with some details.

One problem I see is I'd expect the post-job hooks being also executed in such case.

So @clebergnu please let me know what do you think about this. I'd prefer keeping it here and adding the `post` hook before quitting, or moving the `pre` hook after the `if not self.test_suite:` to not execute it on empty test_suites.